validation/lst1-dl3-crab/make.py

```diff
@@ -0,0 +1,215 @@
+    )
+    flux_points = fpe.run(datasets)
+
+    log.info(f"Writing flux points file to {PATH_RESULTS}")
+    table = flux_points.to_table(sed_type="e2dnde", format="gadf-sed")
+    table.write(PATH_RESULTS / "flux-points.ecsv", overwrite=True)
+
+
+@cli.command("run-analysis", help="Run DL3 analysis validation")
+def run_analysis():
+    """1D analysis of LST-1 Crab observations with energy-dependent directional cuts.
+    
+    It performs the following steps:
+    1. DL3 to DL4 data reduction.
+    2. Spectral model fitting using a LogParabola model.
+    3. Flux points estimation.
+    """
+
+    start_time = time.time()
+
+    log.info("Running 1D spectral analysis for LST-1 Crab Nebula observations.")
+    PATH_RESULTS.mkdir(exist_ok=True, parents=True)
+
+    log.info("Running DL3 to DL4 reduction.")
+    datasets = dl3_to_dl4_reduction(DL3_PATH)
+
+    log.info("Running spectral model fitting.")
+    spectral_model_fitting(datasets)
+
+    log.info("Running flux points estimation.")
+    get_flux_points(datasets)
+
+    end_time = time.time()
+    duration = end_time - start_time
+    log.info(
+        "Validation completed successfully: "
+        "DL3 to DL4 reduction, spectral model fitting, and flux points estimation. "
+        f"Total time taken: {duration:.0f} s ({duration/60:.1f} min)."
+    )
+
+
+def download_file(url, local_filename) -> None:
+    """Download a file from a given URL to a local path."""
+    if Path(local_filename).exists():
+        log.debug(f"{local_filename} already exists, skipping download.")
+        return
+    with requests.get(url, stream=True) as request:
+        request.raise_for_status()
+        with open(local_filename, 'wb') as file:
+            file.write(request.content)
+
+
+def get_data_from_zenodo(zenodo_record_id=ZENODO_URL_LST1_CRAB, dir_path=DL3_PATH) -> None:
+    """Use Zenodo API to download files from a given Zenodo record ID.
+    
+    Parameters
+    ----------
+    zenodo_record_id : str
+        The Zenodo record ID to download files from.
+        Default is LST-1 Crab Zenodo record.
+    dir_path : Path
+        The directory path where files will be downloaded.
+
+    TODO: add checksum validation 
+    """
+    dir_path.mkdir(exist_ok=True, parents=True)
+
+    log.info("Downloading files from Zenodo.")
+
+    zenodo_url = f'https://zenodo.org/api/records/{zenodo_record_id}'
+    response = requests.get(zenodo_url)
+
+    if response.status_code == 200:  # OK status
+        record_data = response.json()['files']
+        file_urls = [entry['links']['self'] for entry in record_data]
+        file_names = [entry['key'] for entry in record_data]
+
+        for file_url, file_name in zip(file_urls, file_names):
+            file_path = dir_path / file_name
+            log.debug(f"Downloading {file_name} from {file_url}...")
+            download_file(file_url, file_path)
+
+    else:
+        log.error("Failed to retrieve the Zenodo record.")
```

Comment:
shouldn't this raise an exception?

Reply:
according to the documentation (https://3.python-requests.org/user/quickstart/#errors-and-exceptions) request will raise errors and exceptions in case status is no okay

Reply:
I raise and error here